patient_data_source.py

```diff
@@ -0,0 +1,214 @@
+from abc import ABC, abstractmethod
+import names
+import warnings
+from datetime import datetime
+from enum import Enum
+from collections.abc import Iterable
+from fhirclient.models.patient import Patient as FHIR_Patient
+from fhirclient.models.observation import Observation as FHIR_Observation
+from fhirclient.models.fhirdate import FHIRDate
+
+class Patient(ABC):
+	class Gender(Enum):
+		male = 0
+		female = 1
+		other = 2
+		unknown = 3
+
+	def get_gender(self) -> Gender:
+		"""Returns the patient's gender. Default implementation return Gender.unkown"""		
+		return Gender.unknown
+
+	def get_identifier_value(self) -> str:
+		"""Returns a id value for external use."""
```

Comment:
typo "a id"

more explanation needed in docstring. "external use" is not sufficiently clear. here is a suggestion:

"Return a custom identifier for the Patient. This is not the ID that will be used internally by the server. This could be used to link the Patient to its point of origin in the data source, for example."

(multi-line docstrings are okay and encouraged!)